DashKit/DashPumpManager.swift

```diff
@@ -571,7 +571,7 @@ public class DashPumpManager: PumpManager {
             switch result {
             case .success(let status):
                 self.mutateState({ (state) in
-                    state.unfinalizedBolus?.cancel(at: self.dateGenerator())
+                    state.unfinalizedBolus?.cancel(at: self.dateGenerator(), withRemaining: status.bolusUnitsRemaining)
```

Comment:
`self.dateGenerator()` is obviously generating a date, but doesn't give any context to that date. Is this date `Now` or part or future or something else? Should it provide more details?

Reply:
It's an injectable replacement for Date(). So yes, typically it would be "now", but can be injected to provide other contexts for testing.